src/helpers/similarity.ts

```diff
@@ -0,0 +1,105 @@
+import { getLogger } from "../bindings";
+import axios, { AxiosError } from "axios";
+import { ajv } from "../utils";
+import { Static, Type } from "@sinclair/typebox";
+import { backOff } from "exponential-backoff";
+import { Issue } from "../types";
+
+export const extractImportantWords = async (issue: Issue): Promise<string[]> => {
+  const res = await getAnswerFromChatGPT(
+    "",
+    `${
+      process.env.CHATGPT_USER_PROMPT_FOR_IMPORTANT_WORDS ||
+      "I need your help to find important words (e.g. unique adjectives) from github issue below and I want to parse them easily so please separate them using #(No other contexts needed). Please separate the words by # so I can parse them easily. Please answer simply as I only need the important words. Here is the issue content.\n"
+    } '${`Issue title: ${issue.title}\nIssue content: ${issue.body}`}'`,
+    parseFloat(process.env.IMPORTANT_WORDS_AI_TEMPERATURE || "0")
+  );
+  if (res === "") return [];
+  return res.split(/[,# ]/);
+};
+
+export const measureSimilarity = async (first: Issue, second: Issue): Promise<number> => {
+  const res = await getAnswerFromChatGPT(
+    "",
+    `
+      ${(
+        process.env.CHATGPT_USER_PROMPT_FOR_MEASURE_SIMILARITY ||
+        'I have two github issues and I need to measure the possibility of the 2 issues are the same content (I need to parse the % so other contents are not needed and give me only the number in %).\n Give me in number format and add % after the number.\nDo not tell other things since I only need the number (e.g. 85%). Here are two issues:\n 1. "%first%"\n2. "%second%"'
+      )
+        .replace("%first%", `Issue title: ${first.title}\nIssue content: ${first.body}`)
+        .replace("%second%", `Issue title: ${second.title}\nIssue content: ${second.body}`)}`,
```

Comment:
This is an unusual, but acceptable, replace mechanism.

Why not use string literals?

```typescript
const myVariable = "testing";
const myStringLiteral = `my string literal ${myVariable}`;
// "my string literal testing"
```

Reply:
I agree, it would be better if it's done the same way as [here](https://github.com/ubiquity/ubiquibot/pull/608/files#diff-289ca7c6ad34c7929d5662f47d65edd1bbd87b256689fd2cd475d61153f10cb1R14)